core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -803,7 +803,7 @@ class ZkUtils(val zkClient: ZkClient,
         createParentPath(BrokerSequenceIdPath, acls)
         try {
           zkClient.createPersistent(BrokerSequenceIdPath, "", acls)
-          0
+          1
```

Comment:
When we create the znode, the version will be 0 and the next call to set data will increment the version to 1. If you make this change, then getSequenceId will return 1 upon creation and 1 again the second time. This call shouldn't really return a number twice, no?

If we need to start from 1, then maybe you need to also change the line above from ` stat.getVersion` to `stat.getVersion + 1`.